sdk/include/opentelemetry/sdk/metrics/export/metric_producer.h

```diff
@@ -31,14 +32,40 @@ namespace metrics
  */
 struct ScopeMetrics
 {
-  const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope_;
+  const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope_ = nullptr;
```

Comment:
Consider sending separate PRs for the SDK fix and the Prometheus exporter changes.

Reply:
Prometheus exporter do not use `resource` in `ResourceMetrics` before and I'm not sure if there will be any problems when prometheus exporter use it without initialize it to a valid value.